src/lib/update-alternatives/UI/main_dialog.rb

```diff
@@ -37,25 +42,38 @@ def dialog_content
     end
 
     def show_handler
-      AlternativesDialog.new.run
+      open_alternatives_dialog
     end
 
     def alternatives_table_handler
-      AlternativesDialog.new.run
+      open_alternatives_dialog
+    end
+
+    def open_alternatives_dialog
+      alternative_index = Yast::UI.QueryWidget(Id(:alternatives_table), :CurrentItem)
+      AlternativesDialog.new(@alternatives_list[alternative_index]).run
     end
 
     def create_table
       Table(
         Id(:alternatives_table),
         Opt(:notify),
         Header(_("Name"), _("Actual alternative"), _("Status")),
-        [
-          Item(Id(:java), "java", "/usr/lib64/jvm/jre-1.8.0-openjdk/bin/java", _("auto")),
-          Item(Id(:an_alternative_name), "an alternative name", "Alternative_path", _("manual"))
-        ]
+        map_alternatives_items
       )
     end
 
+    def map_alternatives_items
+      @alternatives_list.map do |alternative|
+        Item(
+          Id(@alternatives_list.find_index(alternative)),
```

Comment:
in fact you can also use with benefit that ruby have iterators, which can be later modified. so if you use `each_with_index.map` then you can get index as argument. See my example in IRB session:

```
["a", "b", "c"].each_with_index.map { |e, i| e + i.to_s }
=> ["a0", "b1", "c2"]
```

If it is not clear, do not hesitate to ask, as chained iterators is a slightly advanced topic.